src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -47,6 +48,9 @@ internal abstract partial class XmlRawWriter : XmlWriter
         // namespace resolver
         protected IXmlNamespaceResolver? _resolver;
 
+        private char[] _primitivesBuffer = new char[36];
```

Comment:
This fits to guid, but if you know bigger types that doesn't fit - we can increase it and always succeed writing primitives from first time.